### PR TITLE
Multi deck support

### PR DESCRIPTION
**Pull request by sciencemanx**
This PR adds multideck support with the proposed solution from issue https://github.com/kerrickstaley/genanki/issues/18#issuecomment-443016527.

### CODE REVIEW

**Comment by sciencemanx** [comment]
Bump 😄

**Comment by sciencemanx** [comment]
@kerrickstaley -- Any chance you could review? The changes are actually pretty small -- shouldn't take too long.

File: genanki/deck.py [hunk 1]
```diff
@@ -13,22 +13,52 @@ def add_note(self, note):
...
+          0
+      ],
+      'timeToday': [
+        163,
+        23598
+      ],
+      'usn': -1
+    }
+
+  def write_to_db(self, cursor, now_ts):
```
Comment (kerrickstaley) [inline genanki/deck.py]: Please retain the `deck_id` and `name` checks.

File: genanki/deck.py [hunk 2]
```diff
@@ -13,22 +13,52 @@ def add_note(self, note):
...
-    self.models[model.model_id] = model
-
-  def write_to_db(self, cursor, now_ts):
-    if not isinstance(self.deck_id, int):
-      raise TypeError('Deck .deck_id must be an integer, not {}.'.format(self.deck_id))
-    if not isinstance(self.name, str):
-      raise TypeError('Deck .name must be a string, not {}.'.format(self.name))
+    self.models[model.id] = model
 
+  def update_models(self):
```
Comment (kerrickstaley) [inline genanki/deck.py]: Make this private (preface with underscore). Although I'm not sure it really warrants splitting out a method anyway.

File: genanki/model.py [hunk 3]
```diff
@@ -1,11 +1,12 @@
 from cached_property import cached_property
 from copy import copy
 import pystache
+import time
 import yaml
 
 class Model:
-  def __init__(self, model_id=None, name=None, fields=None, templates=None, css=''):
-    self.model_id = model_id
+  def __init__(self, id=None, name=None, fields=None, templates=None, css=''):
```
Comment (kerrickstaley) [inline genanki/model.py]: As above, please change this back to `model_id`.

File: tests/test_genanki.py [hunk 4]
```diff
@@ -210,26 +224,6 @@ def test_media_files(self):
     missing, unused, invalid = self.col.media.check()
     assert set(missing) == {'missing.mp3', 'missing.jpg'}
 
-  def test_write_deck_without_deck_id_fails(self):
```
Comment (kerrickstaley) [inline tests/test_genanki.py]: Please add these tests back.

File: genanki/db.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  models_json, = cursor.execute('SELECT models FROM col').fetchone()
+  assert models_json is not None
+
+  models = json.loads(models_json)
+  models[str(model.id)] = model.to_dict(deck_id)
+
+  new_models_json = json.dumps(models)
+  cursor.execute('UPDATE col SET models = ?', (new_models_json,))
+
+def add_note(cursor, note):
```
Comment (kerrickstaley) [inline genanki/db.py]: Remove unused functions.

**Comment by marccarre** [comment]
This is awesome! Thanks a lot! 🙂I really look forward to having this merged & released as I'm also facing #18. @sciencemanx, when do you think you will have a chance to address @kerrickstaley's comments?

**Comment by kerrickstaley** [comment]
I fixed this bug in https://github.com/kerrickstaley/genanki/commit/53bc7b4e1be3f19af022ef293faa0040bc5c35ee, and I've pushed the new version (0.6.4) to PyPI.

I ended up rewriting the patch because this PR refactors things in a way that doesn't match the existing code style, and which may make it harder to maintain the code long-term.

Sorry for the delay on fixing this, and thanks for your help!

**Pull request closed without merging**